src/binder/binder_util.cpp

```diff
@@ -62,22 +62,21 @@ void BinderUtil::CheckAndTryPromoteType(const common::ManagedPointer<parser::Con
         // TODO(WAN): A bit stupid to take the string view back into a string.
         switch (desired_type) {
           case type::TypeId::DATE: {
-            auto parsed_date = util::TimeConvertor::ParseDate(std::string(str_view));
+            auto parsed_date = execution::sql::Date::FromString(std::string(str_view));
```

Comment:
It looks like FromString just takes a string_view. Can we avoid the string instantiation now?